src/main/java/teammates/common/util/ApiRequest.java

```diff
@@ -0,0 +1,42 @@
+package teammates.common.util;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.HttpResponseException;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+
+/**
+ * A utility class to execute an API request and return the response.
+ */
+public final class ApiRequest {
+
+    private ApiRequest() {
+        // Utility class
+        // Intentional private constructor to prevent instantiation.
+    }
+
+    /**
+     * Returns the API HTTP response string.
+     * @param uri The URI containing the request URL and request parameters.
+     * @return the HTTP response string after executing the API request
+     */
+    public static String execute(URI uri) throws IOException, NullPointerException {
+        HttpUriRequest request = new HttpGet(uri);
+        HttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);
```

Comment:
When interact with third party, we cannot assume it will always perform as expected. Can we set a timeout value here to be safe?